lib/commands/source.js

```diff
@@ -13,13 +13,15 @@ const commands = {
       return await this.executeAtom('execute_script', [script, []]);
     }
 
-    if ((await this.settings.getSettings()).useJSONSource) {
-      const srcTree = await this.mobileGetSource('json');
+    const settings = await this.settings.getSettings();
+    const excludedAttributes = settings.pageSourceExcludedAttributes;
+    const hasExcludedAttributes = _.isString(excludedAttributes) && !_.isEmpty(excludedAttributes);
+
+    if (settings.useJSONSource) {
+      const srcTree = await this.mobileGetSource('json', hasExcludedAttributes ? excludedAttributes : undefined);
```

Comment:
Please also:
- bump the minimum WDA version in package.json
- update the documentation for the corresponding setting at https://github.com/appium/appium-xcuitest-driver/blob/master/docs/reference/settings.md

Reply:
done